backend/common/server/config.py

```diff
@@ -85,19 +91,30 @@ def before_request():
         )
 
     def after_request(response: Response):
+        g.track_memory = False
+        g.memory_thread.join()  # Ensure the memory tracking thread has finished
+        total_time = time.time() - g.start_time
+        peak_memory = max(g.memory_usage) if g.memory_usage else 0
         app.logger.info(
             dict(
                 type="RESPONSE",
                 details=dict(
                     status_code=response.status_code,
                     content_length=response.content_length,
-                    response_time=time.time() - g.start,
+                    response_time=total_time,
+                    peak_memory_usage=peak_memory,
                 ),
             )
         )
-        response.headers["X-Request-Id"] = get_request_id()
+        response.headers.add("X-Request-Id", get_request_id())
+        response.headers.add("X-Total-Process-Time", f"{total_time:.3f}s")
+        response.headers.add("X-Peak-Memory-Usage", f"{peak_memory:.3f} MiB")
         return response
 
+    def _track_memory_usage():
+        while g.track_memory:
+            g.memory_usage.append(memory_usage(-1, interval=0.25, timeout=0.25)[0])
```

Comment:
With the current interval, what happens if the request is faster than this sample rate?

Reply:
If the request is faster than 250ms, then it almost certainly has a small memory footprint. As of the most recent commit, peak_memory would just be set to 0 if there were no samples:
```
peak_memory = max(g.memory_usage) if g.memory_usage else 0
```